### PR TITLE
'Rename VM' feature is missing on vm settings modal

### PR DESCRIPTION
**Pull request by hakankaradis**
User will be able to update machine name from virtual machine tab.
## Description

When user focus on any machine name on virtual machine, we will give user ability to update machine label and after user blur input box or hit enter we will update the machine name.
## Motivation and Context

#8643
## How Has This Been Tested?

Go to virtual machine tab, edit any machine name then blur input box or hit enter. You will see updated machine name from both sidebar and the virtual machine tab.
## Screenshots (if appropriate):

http://recordit.co/rIBYZ2eZsL
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: client/home/lib/virtualmachines/components/machineslist/listitem.coffee [hunk 1]
```diff
@@ -122,17 +155,19 @@ module.exports = class MachinesListItem extends React.Component
...
-        <div
-          className="MachinesListItem-machineLabel #{@props.machine.getIn ['status', 'state']}"
-          onClick={@bound 'toggle'}>
-          {@props.machine.get 'label'}
+      <header className='MachinesListItem-header'>
+        <div className="MachinesListItem-machineLabel #{@props.machine.getIn ['status', 'state']}">
+          <input
+            ref='inputbox'
+            value={@state.machineLabel}
+            className="kdinput text template-title autogrow"
```
Comment (sinan) [inline client/home/lib/virtualmachines/components/machineslist/listitem.coffee]: `atuogrow`?

**Comment by hakankaradis** [comment]
@sinan here is the final screencast http://recordit.co/PaZqKOBhQ7

**Comment by hakankaradis** [comment]
@sinan ping

File: client/stack-editor/lib/editor/index.coffee [hunk 2]
```diff
@@ -272,10 +273,14 @@ module.exports = class StackEditorView extends kd.View
         { changeTemplateTitle } = EnvironmentFlux.actions
         changeTemplateTitle stackTemplate?._id, e.target.value
 
+    @header.addSubView @editName = new CustomLinkView
+      cssClass: 'edit-name'
+      title: 'Edit Name'
+      click : =>
```
Comment (sinan) [inline client/stack-editor/lib/editor/index.coffee]: `click : @inputTitle.bound 'setFocus'`

File: client/stack-editor/lib/editor/index.coffee [hunk 3]
```diff
@@ -288,6 +293,13 @@ module.exports = class StackEditorView extends kd.View
       @inputTitle.prepareClone()
       @inputTitle.resize()
 
+    @inputTitle.on 'blur', =>
```
Comment (sinan) [inline client/stack-editor/lib/editor/index.coffee]: `@inputTitle.on 'blur', @editName.bound 'show'`

**Comment by sinan** [comment]
minor comments, can be sent in with any other PR later. 👍

**Pull request merged**